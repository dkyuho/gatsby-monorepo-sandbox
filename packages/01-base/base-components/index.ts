// 1. colours and siteWideGaps must be defined first
export { default as coloursBase } from './src/tokens/colours-base'
export { default as coloursTheme } from './src/tokens/colours-theme'
export { default as siteWideGaps } from './src/tokens/site-wide-gaps'
// 2. mixins used in tokens must be defined next
export {
  primaryFontStack,
  primaryFontStackMedium,
  primaryFontStackBold,
  secondaryFontStack,
  secondaryFontStackMedium,
  secondaryFontStackBold,
} from './src/mixins/mixins-font-stacks'
// 3. siteWide must be defined next because values are used in headerTokens
export { default as buttonsLinksTokens } from './src/tokens/buttons-links-tokens'
export { default as siteWide } from './src/tokens/site-wide'
export { default as linkReset } from './src/mixins/mixins-links-reset'
export {
  buttonPrimary,
  buttonSecondary,
  buttonOutline,
  buttonSizeLarge,
  buttonSizeSmall,
  buttonCloseSecondary,
  buttonStickyLeftRight,
} from './src/mixins/mixins-buttons'
export { default as buttonReset } from './src/mixins/mixins-buttons-reset'
export { default as svgFriendlyColour } from './src/functions/svg-friendly-colour'
export { default as headerThemeVariables } from './src/constants/constants-header-theme-variables'
// 4. tokens must be defined after siteWide
export {
  bounceInLeft,
  fadeIn,
  fadeOut,
  fadeOutIn,
  fadeInMoveY,
  flipInY,
  zoomIn,
  grow150FadeOut,
  grow200FadeOut,
  growFadeinMoveY,
  growSlowFadeIn,
  growSlowFadeInFast,
  LeftIn,
  LeftIn2,
  overflowIn,
  pulse,
  rotateBanner,
  slideIn,
  spin,
} from './src/tokens/animations'
export { default as breakpoints } from './src/tokens/breakpoints'
export { default as buttonsTokens } from './src/tokens/buttons-tokens'
export { default as cardWrapperTokens } from './src/tokens/card-wrapper-tokens'
export { default as cardWrapper } from './src/mixins/mixins-card-wrapper'
export { default as zindex } from './src/tokens/zindex'
export { default as locationsTokens } from './src/tokens/locations-tokens'
export { default as modalTokens } from './src/tokens/modal-tokens'
export { default as pagesTokens } from './src/tokens/pages-tokens'
export { default as teamMembersTokens } from './src/tokens/team-members-tokens'
export { default as testimonialsTokens } from './src/tokens/testimonials-tokens'
// 5. Many of the following mixins import tokens and must be defined after tokens
export { default as lazyFadeIn } from './src/mixins/mixins-animations'
export { gridMaker } from './src/mixins/mixins-columns-grids'
export {
  mediaQueryMinPx,
  mediaQueryMin,
  mediaQueryMax,
  mediaQueryBetween,
  mediaQueryHover,
} from './src/mixins/mixins-media-queries'
export { default as modalBtnCloseWrapper } from './src/mixins/mixins-modals'
export { siteWidePadding } from './src/mixins/mixins-paddings-site-wide'
export { tabItemLarge, tabItemSmall, tabItemActive, tabbedList, tabbedItem } from './src/mixins/mixins-tabs'
// Functions
export { sortArrayByKey } from './src/functions/form-helpers'
export { default as getImageFileType } from './src/functions/get-image-file-type'
export { default as getSetSortedByKey } from './src/functions/get-set-sorted-by-key'
export { weekdaysArray, getWeekday, getMonth } from './src/functions/get-weekday-month-name'
export { isNotEmptyArray, isEmptyObject } from './src/functions/validate-values'
export { default as hourTo12HrFormat } from './src/functions/convert-hour'
export { default as removeDuplicateObject } from './src/functions/remove-duplicate-object'
export { default as stripNonNumerics } from './src/functions/strip-non-numerics'
export { encodeEmail, decodeEmail } from './src/functions/obfuscate-email'
export { default as getActiveHolidayEvents } from './src/functions/get-active-holiday-events'
export { default as isAbsoluteUrl } from './src/functions/is-absolute-url'
// Constants
export {
  baseCarouselSettings,
  stackedMultiCarouselSettings,
  rowCarouselSettings,
  homeHeroCarouselSettings,
  homePromosCarouselSettings,
  stockWishlistCarouselSettings,
  testimonialsCarouselSettings,
  vfTabbedCarouselSettings,
  sdKeyImageCarouselSettings,
  sdKeyImageSmallCarouselSettings,
  sdSimilarVehicleCarouselSettings,
  stockCarouselSettingsTwoCol,
  stockCarouselSettingsOneCol,
  blogCarouselSettings,
  vehicleBannerCarouselSettings,
  buildBuyConfiguratorCarouselSettings,
  brandLogoLinksCarouselSettings,
} from './src/constants/constants-carousel-settings'
// Static Queries
export { default as staticQueryPrivacy } from './src/static-queries/static-query-privacy'
export { default as staticQueryTestimonials } from './src/static-queries/static-query-testimonials'
export { default as staticQueryLocations } from './src/static-queries/static-query-locations'
export { default as staticQueryTeams } from './src/static-queries/static-query-teams'
// Custom Hooks
export { default as useHasRendered } from './src/hooks/hook-use-has-rendered'
export { default as useIntersectionObserver } from './src/hooks/hook-use-intersection-observer'
export { default as useTabs } from './src/hooks/hook-use-tabs'
// SVG Defs
export { default as Svgs24 } from './src/components/svgs/svgs-24'
export { default as Svgs } from './src/components/svgs/svgs'
export { default as SvgsStock } from './src/components/svgs/svgs-stock'
// Headings
export {
  heading1Styles,
  heading2Styles,
  heading3Styles,
  heading4Styles,
  heading5Styles,
  heading6Styles,
} from './src/components/headings/headings-styles'
export { Heading1, Heading2, Heading3, Heading4, Heading5, Heading6 } from './src/components/headings/headings'

// Elements
export { default as Paragraph } from './src/components/paragraph/paragraph'
export { default as paragraphStyles } from './src/components/paragraph/paragraph-styles'
// Links
export { default as LinkBasic } from './src/components/links/link-basic'
export { linkBasicStyles, linkBasicExternalStyles } from './src/components/links/link-basic-styles'
// Wrappers
export { default as ContainerPadded } from './src/components/wrappers/container-padded'
export { default as containerPaddedStyles } from './src/components/wrappers/container-padded-styles'
export { default as Container } from './src/components/wrappers/container'
export { default as containerStyles } from './src/components/wrappers/container-styles'
export { default as Main } from './src/components/wrappers/main'
export { default as mainStyles } from './src/components/wrappers/main-styles'
export { default as Wrapper } from './src/components/wrappers/wrapper'
export { default as WrapperFlex } from './src/components/wrappers/wrapper-flex'
export { default as wrapperFlexStyles } from './src/components/wrappers/wrapper-flex-styles'
export { default as WrapperPadded } from './src/components/wrappers/wrapper-padded'
export { default as wrapperPaddedStyles } from './src/components/wrappers/wrapper-padded-styles'
// Markdown
export { default as ReactMarkdown } from './src/components/markdown/react-markdown'
// Types
export { default as ButtonsTokens } from './src/types/buttons-tokens-types'
export { default as ButtonsLinksTokens } from './src/types/buttons-links-tokens-types'
export { default as CardWrapperTokens } from './src/types/card-wrapper-tokens-types'
export { default as ColoursTokens } from './src/types/colours-tokens-types'
export { default as ModalTokens } from './src/types/modal-tokens-types'
export { default as PagesTokens } from './src/types/pages-tokens-types'
export { default as SiteWideTokens } from './src/types/site-wide-tokens-types'
export { default as TeamMembersTokens } from './src/types/team-members-tokens-types'
export { default as TestimonialsTokens } from './src/types/testimonials-tokens-types'
export { default as LocationsTokens } from './src/types/locations-tokens-types'
export type { EmotionStyles } from './src/types/emotion-styles-type'
