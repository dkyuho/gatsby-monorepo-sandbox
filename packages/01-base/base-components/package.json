{
  "name": "base-components",
  "version": "0.0.1",
  "description": "Base theme. It will contain basic components.",
  "main": "index.ts",
  "author": "dkyuho",
  "license": "MIT",
  "private": true,
  "sideEffects": [
    "*.css"
  ],
  "peerDependencies": {
    "gatsby": "^5.11.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0"
  },
  "dependencies": {
    "@emotion/react": "^11.10.8",
    "gatsby-plugin-emotion": "^8.9.0",
    "gatsby-plugin-manifest": "^5.9.0",
    "gatsby-plugin-react-svg": "^3.3.0",
    "gatsby-source-filesystem": "^5.9.0",
    "gatsby-transformer-json": "^5.11.0",
    "react-intersection-observer": "^9.4.3",
    "react-markdown": "8.0.7",
    "rehype-raw": "6.1.1",
    "rehype-sanitize": "5.0.1"
  },
  "devDependencies": {
    "@babel/core": "^7.21.8",
    "@emotion/babel-plugin": "^11.10.8",
    "@emotion/babel-preset-css-prop": "^11.10.0",
    "@emotion/eslint-plugin": "^11.10.0",
    "@emotion/jest": "^11.10.8",
    "@testing-library/jest-dom": "^5.16.5",
    "@testing-library/react": "^14.0.0",
    "@types/react-datepicker": "^4.10.0",
    "@types/react-slick": "^0.23.10",
    "babel-jest": "^29.5.0",
    "babel-loader": "^9.1.2",
    "babel-plugin-remove-graphql-queries": "^5.9.0",
    "babel-preset-gatsby": "^3.9.0",
    "color": "^4.2.3",
    "eslint": "^8.39.0",
    "eslint-plugin-import": "^2.27.5",
    "eslint-plugin-prettier": "^4.2.1",
    "eslint-plugin-react": "^7.32.1",
    "identity-obj-proxy": "^3.0.0",
    "jest": "^29.5.0",
    "prop-types": "^15.8.1",
    "webpack-filter-warnings-plugin": "^1.2.1"
  }
}
