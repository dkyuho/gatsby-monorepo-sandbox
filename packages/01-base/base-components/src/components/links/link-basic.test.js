/**
 * @jest-environment jsdom
 */
// https://jestjs.io/docs/configuration#testenvironment-string
import React from 'react'
import { render } from '@testing-library/react'
import { LinkBasic } from 'base-components'

describe('<LinkBasic />', () => {
  const internalLink = {
    to: '/test-url',
  }
  const externalLink = {
    to: 'www.google.com',
  }
  const customProps = {
    'custom-prop': 'custom-prop',
  }

  const setup = (props) => {
    const wrapper = render(<LinkBasic {...props}>Link Gatsby</LinkBasic>)
    expect(wrapper.container).toMatchSnapshot()
    return { wrapper, gatsbyLink: wrapper.queryByTestId('gatsby-link') }
  }

  it('matches basic snapshot', () => {
    const { gatsbyLink } = setup(internalLink)
    expect(gatsbyLink).toHaveTextContent('Link Gatsby')
    expect(gatsbyLink).toBeEnabled()
  })

  it('matches snapshot with external links', () => {
    const { wrapper, gatsbyLink } = setup(externalLink)
    expect(gatsbyLink).not.toBeInTheDocument()
    const linkElement = wrapper.queryByText('Link Gatsby')
    expect(linkElement).toBeInTheDocument()
    expect(linkElement).toBeEnabled()
  })

  it('matches snapshot with custom props', () => {
    const { gatsbyLink } = setup({ ...internalLink, ...customProps })
    expect(gatsbyLink).toHaveTextContent('Link Gatsby')
    expect(gatsbyLink).toHaveAttribute('custom-prop', customProps['custom-prop'])
  })
})
