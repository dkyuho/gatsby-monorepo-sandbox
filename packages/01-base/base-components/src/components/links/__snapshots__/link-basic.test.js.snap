// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<LinkBasic /> matches basic snapshot 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: 333;
  cursor: pointer;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <a
    class="classLinkBasic emotion-0"
    data-testid="gatsby-link"
    href="/test-url"
  >
    Link Gatsby
  </a>
</div>
`;

exports[`<LinkBasic /> matches snapshot with custom props 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: 333;
  cursor: pointer;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <a
    class="classLinkBasic emotion-0"
    custom-prop="custom-prop"
    data-testid="gatsby-link"
    href="/test-url"
  >
    Link Gatsby
  </a>
</div>
`;

exports[`<LinkBasic /> matches snapshot with external links 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: 333;
  cursor: pointer;
}

.emotion-0:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

<div>
  <a
    class="classLinkBasic emotion-0"
    href="www.google.com"
  >
    Link Gatsby
  </a>
</div>
`;
